File: 프로그래머스/lv0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> sides)
{
    int answerA = 0;
    int answerB = 0;

    sort(sides.begin(), sides.end());

    for (int i = 0; i <= sides[1]; i++)
    {
        if (sides[0] + i >= sides[1])
            answerA++;
    }

    for (int i = sides[1]; i < sides[0] + sides[1]; i++)
    {
        if (sides[0] + sides[1] >= sides[1])
            answerB++;
    }


    return answerA + answerB - 2;
}

File: 프로그래머스/0/120868. 삼각형의 완성조건 （2）/삼각형의 완성조건 （2）.java
import java.util.Arrays;

class Solution {

    public int solution(int[] sides) {
        Arrays.sort(sides);

        if (sides[1] + 1 > sides[0]) {
            return sides[0] + (sides[0] - 1);
        }

        return sides[0];
    }
}